src/main/kotlin/dev/arbjerg/ukulele/features/LeaveOnIdle.kt

```diff
@@ -0,0 +1,78 @@
+package dev.arbjerg.ukulele.features
+
+import dev.arbjerg.ukulele.audio.Player
+import org.springframework.stereotype.Service
+import dev.arbjerg.ukulele.audio.PlayerRegistry
+import dev.arbjerg.ukulele.config.BotProps
+import dev.arbjerg.ukulele.data.GuildPropertiesService
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import net.dv8tion.jda.api.entities.Guild
+import net.dv8tion.jda.api.entities.Member
+import net.dv8tion.jda.api.entities.VoiceChannel
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.scheduling.TaskScheduler
+import java.time.Duration
+import java.util.concurrent.ScheduledFuture
+
+@Service
+class LeaveOnIdle(
+        private val players: PlayerRegistry,
+        private val guildPropertiesService: GuildPropertiesService,
+        private val scheduler: TaskScheduler,
+        private val botProps: BotProps
+) {
+    private val log: Logger = LoggerFactory.getLogger(LeaveOnIdle::class.java)
+    private val timers: MutableMap<Guild, ScheduledFuture<*>> = HashMap()
+
+    fun onVoiceJoin(guild: Guild, channel: VoiceChannel, member: Member) {
+        if (member == guild.selfMember) {
+            log.info("Joined voice channel {} in guild {}", channel.name, guild.name)
+            createTimer(guild)
+        }
+    }
+
+    fun onVoiceLeave(guild: Guild, channel: VoiceChannel, member: Member) {
+        if (member == guild.selfMember) {
+            log.info("Left voice channel {} in guild {}", channel.name, guild.name)
+            removeTimer(guild)
+        }
+    }
+
+    private fun createTimer(guild: Guild) {
+        GlobalScope.launch {
```

Comment:
Why is this within a coroutine?

Reply:
I copied the usage of `getAwait` from https://github.com/freyacodes/ukulele/blob/master/src/main/kotlin/dev/arbjerg/ukulele/jda/CommandManager.kt#L44. I saw that there is a synchronous `get` method on `GuildPropertiesService`, but that itself returns a `Mono`, and I think the only way to resolve the `Mono` synchronously is `block()` which I didn't want to do as it would block the calling thread, which would probably be un-ideal. 

 Is there a different way to invoke asynchronous code from a synchronous context in this case?

Reply:
Ah, didn't see that.

Why does this need to be run synchronously? You can resolve a mono asynchronously with `subscribe(Consumer)`

Reply:
That's fair, I guess it doesn't have to be synchronous. I'll use `subscribe` 👍🏾